lib/base_service.ts

```diff
@@ -56,23 +59,30 @@ export interface Credentials {
   password?: string;
   url?: string;
   iam_access_token?: string;
+  icp_access_token?: string;
   iam_apikey?: string;
   iam_url?: string;
+  authentication_type?: string;
 }
 
 function hasCredentials(obj: any): boolean {
   return (
     obj &&
     ((obj.username && obj.password) ||
       obj.iam_access_token ||
-      obj.iam_apikey)
+      obj.iam_apikey ||
+      obj.icp_access_token)
   );
 }
 
 function isForICP(cred: string): boolean {
   return cred && cred.startsWith('icp-');
 }
 
+function isForICP4D(obj: any): boolean {
+  return obj && (obj.authentication_type === 'icp4d' || obj.icp_access_token);
```

Comment:
This isn't consistent with the README.md where it says:
> In order to use ICP4D authentication, this option [`authentication_type`] **must** be passed in.

Reply:
Oh, good point. I decided after writing the docs that `icp_access_token` is sufficient for determining if `icp4d` is the authentication type. I'll update the docs to reflect that.

Also, I wonder if that variable should be called `icp4d_access_token` for consistency...